### PR TITLE
3.4

### PR DESCRIPTION
**Pull request by ajibarra**

### CODE REVIEW

File: src/Service/Action/CrudAction.php [hunk 1]
```diff
@@ -256,7 +256,7 @@ public function parentIdName()
     protected function _describe()
     {
         $table = $this->table();
```
Comment (steinkel) [inline src/Service/Action/CrudAction.php]: use getTable

File: src/Service/Action/Extension/CrudAutocompleteListExtension.php [hunk 2]
```diff
@@ -45,23 +45,23 @@ public function implementedEvents()
...
+    protected function _autocompleteList(CrudAction $action, Query $query)
     {
         $data = $action->data();
         if (!(is_array($data) && !empty($data['autocomplete_list']))) {
             return $query;
         }
         $query = $query->select([
-            $action->table()->primaryKey(),
-            $action->table()->displayField()
+            $action->table()->getPrimaryKey(),
```
Comment (steinkel) [inline src/Service/Action/Extension/CrudAutocompleteListExtension.php]: use getTable

File: src/Service/Action/Extension/CrudAutocompleteListExtension.php [hunk 3]
```diff
@@ -45,23 +45,23 @@ public function implementedEvents()
...
+            $action->table()->getDisplayField()
```
Comment (steinkel) [inline src/Service/Action/Extension/CrudAutocompleteListExtension.php]: use getTable

File: src/Service/Action/Extension/NestedExtension.php [hunk 4]
```diff
@@ -69,7 +69,7 @@ public function findEntities(Event $event)
      */
     public function findEntity(Event $event)
     {
-        $action = $event->subject();
+        $action = $event->getSubject();
         $query = $event->data['query'];
```
Comment (steinkel) [inline src/Service/Action/Extension/NestedExtension.php]: use getData()

File: src/Service/Action/Extension/NestedExtension.php [hunk 5]
```diff
@@ -91,7 +91,7 @@ public function findEntity(Event $event)
      */
     public function patchEntity(Event $event)
     {
-        $action = $event->subject();
+        $action = $event->getSubject();
         $entity = $event->data['entity'];
```
Comment (steinkel) [inline src/Service/Action/Extension/NestedExtension.php]: getData()

File: src/Service/Action/Extension/PaginateExtension.php [hunk 6]
```diff
@@ -52,7 +52,7 @@ public function implementedEvents()
      */
     public function findEntities(Event $event)
     {
-        $action = $event->subject();
+        $action = $event->getSubject();
         $query = $event->data['query'];
```
Comment (steinkel) [inline src/Service/Action/Extension/PaginateExtension.php]: getData()

File: src/Service/Action/Extension/PaginateExtension.php [hunk 7]
```diff
@@ -111,7 +111,7 @@ protected function _limit(Action $action)
      */
     public function afterFind(Event $event)
     {
-        $action = $event->subject();
+        $action = $event->getSubject();
         $query = $event->data['query'];
```
Comment (steinkel) [inline src/Service/Action/Extension/PaginateExtension.php]: getData()

File: src/Service/Action/Extension/SortExtension.php [hunk 8]
```diff
@@ -53,7 +53,7 @@ public function implementedEvents()
      */
     public function findEntities(Event $event)
     {
-        $action = $event->subject();
+        $action = $event->getSubject();
         $query = $event->data['query'];
```
Comment (steinkel) [inline src/Service/Action/Extension/SortExtension.php]: getData()

File: src/Service/Renderer/JSendRenderer.php [hunk 9]
```diff
@@ -73,7 +73,7 @@ public function accept()
     public function response(Result $result = null)
     {
         $response = $this->_service->response();
-        $response->statusCode($result->code());
+        $response->withStatus($result->code());
```
Comment (steinkel) [inline src/Service/Renderer/JSendRenderer.php]: response immutable

File: src/Service/Renderer/JSendRenderer.php [hunk 10]
```diff
@@ -73,7 +73,7 @@ public function accept()
...
         $response->type('application/json');
```
Comment (steinkel) [inline src/Service/Renderer/JSendRenderer.php]: withType() and immutable

File: src/Service/Renderer/JSendRenderer.php [hunk 11]
```diff
@@ -106,8 +106,8 @@ public function error(Exception $exception)
         }
         $message = $this->_buildMessage($exception);
         $trace = $this->_stackTrace($exception);
-        $response->statusCode((int)$this->errorCode);
-        $response->body($this->_error($message, $exception->getCode(), $data, $trace));
+        $response->withStatus((int)$this->errorCode);
```
Comment (steinkel) [inline src/Service/Renderer/JSendRenderer.php]: immutable

File: src/Service/Renderer/JsonRenderer.php [hunk 12]
```diff
@@ -35,15 +35,14 @@ class JsonRenderer extends BaseRenderer
     public function response(Result $result = null)
     {
         $response = $this->_service->response();
-        $response->statusCode($result->code());
+        $response->withStatus($result->code());
```
Comment (steinkel) [inline src/Service/Renderer/JsonRenderer.php]: immutable

File: src/Service/Renderer/RawRenderer.php [hunk 13]
```diff
@@ -33,9 +33,9 @@ class RawRenderer extends BaseRenderer
     public function response(Result $result = null)
     {
         $response = $this->_service->response();
-        $response->statusCode($result->code());
+        $response->withStatus($result->code());
         $response->type('text/plain');
```
Comment (steinkel) [inline src/Service/Renderer/RawRenderer.php]: withType()

File: src/Service/Renderer/XmlRenderer.php [hunk 14]
```diff
@@ -39,10 +39,10 @@ class XmlRenderer extends BaseRenderer
     public function response(Result $result = null)
     {
         $response = $this->_service->response();
-        $response->statusCode($result->code());
+        $response->withStatus($result->code());
```
Comment (steinkel) [inline src/Service/Renderer/XmlRenderer.php]: immutable

File: src/Service/Service.php [hunk 15]
```diff
@@ -603,7 +605,7 @@ public function respond($result = null)
             $result = $this->result();
         }
         $this->response()
```
Comment (steinkel) [inline src/Service/Service.php]: immutable

File: src/Middleware/RequestHandlerMiddleware.php [hunk 16]
```diff
@@ -40,8 +35,8 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
             'json' => ['json_decode', true],
             'xml' => [[$this, 'convertXml']],
         ];
-		$this->request = RequestTransformer::toCake($request);
-		$this->response = ResponseTransformer::toCake($response);
+		$this->request = $request;
```
Comment (ajibarra) [inline src/Middleware/RequestHandlerMiddleware.php]: .

**Pull request merged**